src/IO/External/InterpolateFromFuka.cpp

```diff
@@ -0,0 +1,155 @@
+    const std::string& info_filename,
+    const tnsr::I<DataVector, 3, Frame::Inertial>& x,
+    [[maybe_unused]] const double interpolation_offset,
+    [[maybe_unused]] const int interp_order,
+    [[maybe_unused]] const double delta_r_rel) {
+  tuples::tagged_tuple_from_typelist<fuka_tags<IdType>> result{};
+  // The FUKA functions are not thread-safe, so we need to lock here.
+  fuka_lock->lock();
+  const auto fuka_data = [&x, &info_filename, &interpolation_offset,
+                          &interp_order, &delta_r_rel]() {
+    if constexpr (IdType == FukaIdType::Bh) {
+      return KadathExportBH(x.begin()->size(), get<0>(x).data(),
+                            get<1>(x).data(), get<2>(x).data(),
+                            info_filename.c_str(), interpolation_offset,
+                            interp_order, delta_r_rel);
+    } else if constexpr (IdType == FukaIdType::Bbh) {
+      return KadathExportBBH(x.begin()->size(), get<0>(x).data(),
+                             get<1>(x).data(), get<2>(x).data(),
+                             info_filename.c_str(), interpolation_offset,
+                             interp_order, delta_r_rel);
+    } else if constexpr (IdType == FukaIdType::Ns) {
+      return KadathExportNS(x.begin()->size(), get<0>(x).data(),
+                            get<1>(x).data(), get<2>(x).data(),
+                            info_filename.c_str());
+    } else if constexpr (IdType == FukaIdType::Bns) {
+      return KadathExportBNS(x.begin()->size(), get<0>(x).data(),
+                             get<1>(x).data(), get<2>(x).data(),
+                             info_filename.c_str());
+    } else if constexpr (IdType == FukaIdType::Bhns) {
+      return KadathExportBHNS(x.begin()->size(), get<0>(x).data(),
+                              get<1>(x).data(), get<2>(x).data(),
+                              info_filename.c_str(), interpolation_offset,
+                              interp_order, delta_r_rel);
+    } else {
+      ERROR("Unrecognized enum value for 'FukaIdType'");
+    }
+  }();
+  // Unlock before copying data to avoid holding the lock for too long.
+  fuka_lock->unlock();
+  // The FUKA functions return tensor components in this order.
+  // See: https://bitbucket.org/fukaws/kadathimporter/src/master/src/importer.h
+  auto& lapse = get<gr::Tags::Lapse<DataVector>>(result);
+  get(lapse) = to_datavector(fuka_data[0]);
+  auto& shift = get<gr::Tags::Shift<DataVector, 3>>(result);
+  get<0>(shift) = to_datavector(fuka_data[1]);
+  get<1>(shift) = to_datavector(fuka_data[2]);
+  get<2>(shift) = to_datavector(fuka_data[3]);
+  auto& spatial_metric = get<gr::Tags::SpatialMetric<DataVector, 3>>(result);
+  get<0, 0>(spatial_metric) = to_datavector(fuka_data[4]);
+  get<0, 1>(spatial_metric) = to_datavector(fuka_data[5]);
+  get<0, 2>(spatial_metric) = to_datavector(fuka_data[6]);
+  get<1, 1>(spatial_metric) = to_datavector(fuka_data[7]);
+  get<0, 2>(spatial_metric) = to_datavector(fuka_data[8]);
```

Comment:
should this be 1,2?

It seems like you are not setting gamma_yz?

Reply:
Thanks, fixed! The FPE happens in Kadath though, so I don't think this was it..

Also I'd like to add a test for this that is enabled only when FUKA is available. Maybe we could put some sample initial data into the FUKA install directory? Do you think some of the data in the FUKA repo would be appropriate, should we generate new data?